src/Raven.Server/Utils/PoolOfThreads.cs

```diff
@@ -322,6 +354,33 @@ internal void SetThreadAffinity(int numberOfCoresToReduce, long? threadMask)
                     // impact and should be very rare
                 }
             }
+
+            private void SetThreadAffinityByPlatform(long affinity)
+            {
+                Debug.Assert(PlatformDetails.RunningOnMacOsx == false);
+
+                if (PlatformDetails.RunningOnPosix == false)
+                {
+                    // windows
+                    _currentProcessThread.ProcessorAffinity = new IntPtr(affinity);
+                    return;
+                }
+
+                if (PlatformDetails.RunningOnLinux)
+                {
+                    SetLinuxThreadAffinity(affinity);
+                }
+            }
+
+            private void SetLinuxThreadAffinity(long affinity)
+            {
+                var ulongAffinity = (ulong)affinity;
+                var result = Syscall.sched_setaffinity((int)_currentUnmangedThreadId, new IntPtr(sizeof(ulong)), ref ulongAffinity);
```

Comment:
`cpu_set_t` is not a type that is known in advanced. Did you check its size on ARM, 32 bits, etc?